src/Microsoft.ServiceFabric.Services.Remoting/V2/Runtime/ExceptionConvertorHelper.cs

```diff
@@ -0,0 +1,165 @@
+
+namespace Microsoft.ServiceFabric.Services.Remoting.V2.Runtime
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Fabric;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Xml;
+    using Microsoft.ServiceFabric.Services.Communication;
+    using Microsoft.ServiceFabric.Services.Remoting.FabricTransport.Runtime;
+
+    internal class ExceptionConvertorHelper
+    {
+        private static readonly string TraceEventType = "ExceptionConvertorHelper";
+        private IEnumerable<IExceptionConvertor> convertors;
+        private FabricTransportRemotingListenerSettings listenerSettings;
+
+        public ExceptionConvertorHelper(IEnumerable<IExceptionConvertor> convertors, FabricTransportRemotingListenerSettings listenerSettings)
+        {
+            this.convertors = convertors;
+            this.listenerSettings = listenerSettings;
+        }
+
+        public ServiceException ToServiceException(Exception originalException, int currentDepth)
+        {
+            ServiceException serviceException = null;
+            foreach (var convertor in this.convertors)
+            {
+                try
+                {
+                    if (convertor.TryConvertToServiceException(originalException, out serviceException))
+                    {
+                        if (++currentDepth > this.listenerSettings.RemotingExceptionDepth)
+                        {
+                            break;
+                        }
+
+                        var innerEx = convertor.GetInnerExceptions(originalException);
+                        if (innerEx != null && innerEx.Length > 0)
+                        {
+                            serviceException.ActualInnerExceptions = new List<ServiceException>();
+                            int currentBreadth = 0;
+                            foreach (var inner in innerEx)
+                            {
+                                if (++currentBreadth > this.listenerSettings.RemotingExceptionDepth)
+                                {
+                                    break;
+                                }
+
+                                serviceException.ActualInnerExceptions.Add(this.ToServiceException(inner, currentDepth));
+                            }
+                        }
+
+                        break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    ServiceTrace.Source.WriteWarning(
+                       TraceEventType,
+                       "Failed to convert ActualException({0}) to ServiceException : Reason - {1}",
+                       originalException.GetType().Name,
+                       ex);
+                }
+            }
+
+            return serviceException;
+        }
+
+        public ServiceException ToServiceException(Exception originalException)
+        {
+            return this.ToServiceException(originalException, 1);
```

Comment:
Why aren't we using the default depth as defined in _FabricTransportRemotingListenerSettings_ instead of hardcoding 1 here?

Reply:
Thats the current depth. In line 38, we check if the current depth is greater than max depth.